packages/exchange-io-erc1888/src/deposit-watcher/deposit-watcher.service.ts

```diff
@@ -2,22 +2,20 @@ import { Contracts } from '@energyweb/issuer';
 import { getProviderWithFallback } from '@energyweb/utils-general';
 import { Injectable, Logger, OnModuleInit } from '@nestjs/common';
 import { ConfigService } from '@nestjs/config';
+import { EventBus } from '@nestjs/cqrs';
 import { ModuleRef } from '@nestjs/core';
 import { Contract, ethers, providers } from 'ethers';
 import moment from 'moment';
 
 import {
     IExchangeConfigurationService,
-    IExternalDeviceService,
-    AccountService,
-    CreateAskDTO,
-    OrderService,
-    TransferStatus,
-    TransferService
+    TransferService,
```

Comment:
This can be removed I guess.

Reply:
The transfer service is needed for below code block. i.e to get the confirmation block. Please confirm.    

 const blockNumber = await this.transferService.getLastConfirmationBlock();

        this.logger.debug(`Starting from block ${blockNumber}`);

        this.provider.resetEventsBlock(blockNumber);
        this.provider.on(
            {
                address: this.registryAddress,
                topics
            },
            (event: providers.Log) => this.processEvent(event)
        );